sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,637 @@
+                    "length as the list of estimators. Provided {} methods "
+                    "for {} estimators."
+                    .format(len(self.predict_method), len(self.estimators))
+                )
+            predict_method = self.predict_method
+
+        # Fit the base estimators on the whole training data. Those
+        # base estimators will be used in transform, predict, and
+        # predict_proba. They are exposed publicly.
+        self.estimators_ = Parallel(n_jobs=self.n_jobs)(
+            delayed(_parallel_fit_estimator)(clone(est), X, y, sample_weight)
+            for est in estimators_ if est not in (None, 'drop')
+        )
+
+        self.named_estimators_ = Bunch()
+        est_fitted_idx = 0
+        for name_est, org_est in zip(names, estimators_):
+            if org_est not in (None, 'drop'):
+                self.named_estimators_[name_est] = self.estimators_[
+                    est_fitted_idx]
+                est_fitted_idx += 1
+
+        # To train the meta-classifier using the most data as possible, we use
+        # a cross-validation to obtain the output of the stacked estimators.
+
+        # To ensure that the data provided to each estimator are the same, we
+        # need to set the random state of the cv if there is one and we need to
+        # take a copy.
+        cv = check_cv(self.cv)
+        if hasattr(cv, 'random_state') and cv.random_state is None:
+            cv.random_state = np.random.RandomState()
+
+        self.predict_method_ = [
+            self._method_name(name, est, meth)
+            for name, est, meth in zip(names, estimators_, predict_method)
+        ]
+
+        predictions = Parallel(n_jobs=self.n_jobs)(
+            delayed(cross_val_predict)(clone(est), X, y, cv=deepcopy(cv),
+                                       method=meth, n_jobs=self.n_jobs,
+                                       verbose=self.verbose)
+            for est, meth in zip(estimators_, self.predict_method_)
+            if est not in (None, 'drop')
+        )
+
+        # Only not None or not 'drop' estimators will be used in transform.
+        # Remove the None from the method as well.
+        self.predict_method_ = [
+            meth for (meth, est) in zip(self.predict_method_, estimators_)
+            if est not in (None, 'drop')
+        ]
+
+        X_meta = self._concatenate_predictions(predictions)
+        self.final_estimator_.fit(X_meta, y)
+
+        return self
+
+    def transform(self, X):
+        """Return class labels or probabilities for X for each estimator.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix} of shape (n_samples, n_features)
+            Training vectors, where `n_samples` is the number of samples and
+            `n_features` is the number of features.
+
+        Returns
+        -------
+        y_preds : ndarray of shape (n_samples, n_estimators)
```

Comment:
Not true for the multi-class case.

Reply:
you probably want different doc strings for the regressor and classifier, I think.